fix(conversation): #WB-3444, unwanted server requests issues

jcbe-ode opened the pull request:
# Description

Cette PR vise à 
* éviter les requêtes vers `GET /conversation/api/folders/undefined/messages`,
* uniformiser le routage,
* renforcer le typage par endroit,
* à utiliser `useSelectedFolders()` en préférence à `useParams()` (qui ne vérifie pas la casse du dossier)

## Fixes

WB-3444

## Type of change

Please check options that are relevant.

- [ ] Chore (PATCH)
- [ ] Doc (PATCH)
- [X] Bug fix (PATCH)
- [ ] New feature (MINOR)

## Which packages changed?

Please check the name of the package you changed

- [ ] admin
- [ ] app-registry
- [ ] archive
- [ ] auth
- [ ] cas
- [ ] common
- [ ] communication
- [X] conversation
- [ ] directory
- [ ] feeder
- [ ] infra
- [ ] portal
- [ ] session
- [ ] test
- [ ] tests
- [ ] timeline
- [ ] workspace

## Tests

1. Describe here the tests you performed
2. Step by step
3. With expected results

# Reminder

- Security flaws
- Performance impacts (think bulk !)
- Unit tests were replayed
- Unit tests were added and/or changed
- I have updated the reminder for the version including my modifications

- [ ] All done ! :smiley:

jcbe-ode committed:
feat(conversation): WB-3466, React migration (#558)

* feat(conversation): WB-3466 react migration init

* update build react node

* update message

* chore: migration, clean up unnecessary files

* chore: migration, update additional files

* chore: update .gitignore

* chore: update .gitignore and build scripts

* feat(conversation): #WB-3466, GET /conversation/conversation renderView

* chore: some frontend files must not be versioned

* chore: generic build.sh frontend build

---------

Co-authored-by: RomuC <romuald.coulaud@gmail.com>

jcbe-ode committed:
feat(conversation): WB-3630, added 3 /api endpoints under (#559)

* chore: fix build scripts and POMs

* chore: added a buildBackend option to build.sh

* chore: rename gradle.properties files

* feat(conversation): #WB-3630, added GET /conversation/api/* endpoints

* feat(conversation, directory): #WB-3630, implement GET /conversation/api/folders/:folderId/messages

* feat(conversation): #WB-3630, implement GET git /conversation/api/folders

* feat(conversation): #WB-3630, implement GET /conversation/api/messages/{messageId} and cleanup code

* Revert "chore: rename gradle.properties files"

This reverts commit afa066769a2682a542446cdbbfa6a30ce6234807.

* chore: feedback from PR

* fix: feedback from PR

jcbe-ode committed:
ci: fix build scripts

jcbe-ode committed:
feat(conversation): #WB-3633, services, queries, routing and basic pages of React app (#562)

* feat(conversation): #WB-3633, initial commit

* wip

* wip

* wip

* fix: rename routes

* wip

* fix: lint error

* fix: redirection for hash-based folders URLs

* feat: create folder and message API service factories

* wip: folder and message queries

* wip: message queries + simplifications

* fix: feedback from PR

* fix: feedback from PR

* fix: feedback from PR

* feat: folderService tests

* feat: messageService tests

jcbe-ode committed:
feat(conversation): #WB-3467, additional renderView of the new web frontend (#563)

jcbe-ode committed:
chore(conversation): upgrade to frontend framework 2.0

jcbe-ode committed:
feat(conversation): #WB-3428, folders menu navigation (#566)

* feat(conversation): #WB-3428, left menu navigation

* wip

* wip

* Desktop Menu with counters

* Wip: Mobile Menu with counters

* wip

* feat: Mobile Menu with counters and navigation

* wip

* feat(conversation): actions on folders

* fix: feedback from PR

jcbe-ode committed:
fix(conversation): #WB-3630, added folder_id, trashed and unread fields, when getting a message in details in API v1+

Romu-C committed:
feat(conversation): #WB-3426 #WB-3462 #WB-3735 message list with actions, search an… (#569)

* feat: #WB-3426 #WB-3462 #WB-3735 message list with actions, search and filter

jcbe-ode committed:
feat(conversation): #WB-3434, progress bar component,  displaying the used storage space compared to the allowed quota (#571)

* feat(conversation): #WB-3434, display free storage space

* wip

* wip: background + animation ?

* final commit

Romu-C committed:
chore: fix unused import

Romu-C committed:
chore: update baseUrl

Romu-C committed:
fix(conversation): #WB-3426 search filter messages list

jcbe-ode committed:
fix(conversation): #WB-3428, use zustand for storing folders tree (#573)

* fix(conversation): #WB-3428, use zustand for storing folders tree

* fix(conversation): #WB-3428, use zustand instead of route loader data directly

* chore: fix imports preventing building the app correctly

jcbe-ode committed:
fix(conversation): #WB-3434, adjust progress bar color when used storage space is > 70%

jcbe-ode committed:
fix(conversation): #WB-3467,  (#576)

* feat(conversation): #WB-3467, routing corrected

* fix: remaining routes

* fix: remaining routes

* fix: remaining routes + tests

* fix: bad path detection

* fix: config files and default router

* feat: refined new URLs format

jcbe-ode committed:
fix(conversation): #WB-3428, #WB-3434, feedback from Design and QA

Romu-C committed:
fix(conversation): #WB-3426 refacto tests and small bug fix (#579)

* fix(conversation): #WB-3426 refacto tests and issues

* format + remove unused file

* feedback JC

jcbe-ode committed:
feat(conversation): #WB-3451, create folder modal (#580)

* feat(conversation): #WB-3451, create folder modal

* wip: replace Select by Dropdown

* wip

* fix: pending state

* feat: minimal test

* feat: update i18n file

* fix: optimistic UI + minor refactoring

Romu-C committed:
fix(conversation): WB-3635 outbox empty screen + missing translation

jcbe-ode committed:
fix(conversation): build folders tree without trashed ones

Romu-C committed:
fix: #WB-3426 ui, badge navbar

jcbe-ode committed:
feat(conversation): #WB-3455, trash folder modal (#581)

* feat(conversation): #WB-3455, trash a folder

* refacto: remove foldersTree from store

* feedback from PR

* feedback from PR

jcbe-ode committed:
feat(conversation): #WB-3454, rename folder modal (#584)

Romu-C committed:
fix(conversation): #WB-3426 attachment icon + checkbox

jcbe-ode committed:
feat(conversation): #WB-3451, tests for the create folder modal

Romu-C committed:
fix(conversation): #WB-3426 message model hasAttachement

jcbe-ode committed:
fix(conversation): #WB-3428, remove drag'n'drop of folders

jcbe-ode committed:
feat(conversation): #WB-3454, tests for the rename folder modal (#585)

mariusode committed:
Feat wb 3686 content transformation (#583)

* chore: prepare next iteration

* fix(workspace): #WB-2928, transfering documents to public space requires being able to read them

* chore: prepare next iteration

* fix(workspace): #WB-2928, transfering documents to public space requires being able to read them

* chore: prepare next iteration

* chore: prepare next iteration

* fix(workspace): #WB-2928, transfering documents to public space requires being able to read them

* chore: prepare next iteration

* fix(workspace): #WB-2928, transfering documents to public space requires being able to read them

* chore: prepare next iteration

* feat: WB-3686, rich content transformation

* fix: WB-3639, fix index creation

* fix: WB-3686, integration tests and fixes

* feat: WB-3686, fix PR reviews

* feat: WB-3686, uniform message_id foreign key across tables

* feat: WB-3686, fix argument case

---------

Co-authored-by: Benjamin Perez <benjamin.perez@web-education.net>
Co-authored-by: Jean-Christophe <jean-christophe.benoit@opendigitaleducation.com>

Romu-C committed:
feat: #WB-3768 clean index html

Romu-C committed:
feat(conversation): #WB-3823 navigate with keyboard + rename component for more lisibility.

mariusode committed:
fix: WB-3686, fix issue with content_version column name

Romu-C committed:
test(conversation): #WB-3426 test for message preview component

evanthonyly committed:
feat(conversation): #WB-3815 move to trash button (#586)

* chore: add PUT /conversation/trash route to the proxy configuration

* feat(conversation): #WB-3815 move to trash button wip

* Remove message from current page when move to trash

* Update header inbox unread counter

* Show toaster after message move to trash

* Test of move to trash

---------

Co-authored-by: Jean-Christophe <jean-christophe.benoit@opendigitaleducation.com>

Romu-C committed:
feat(conversation): #WB-3825 message attachments (#589)

Romu-C committed:
fix: #WB-3815 build error

jcbe-ode committed:
feat(conversation): #WB-3428, implement design for the tree of folder… (#590)

* feat(conversation): #WB-3428, implement design for the tree of folders with a depth of 3

* fix: adapt new ApiController to trees of depth 3+

* fix: max tree depth

* fix: text color and weight

* fix: comparators with java Set

mariusode committed:
feat: WB-3821, requesting conversation-specific content transformations (#587)

* feat: WB-3821, requesting conversation-specific content transformations

* feat: WB-3821, fix after review

Romu-C committed:
feat(conversation): #WB-3855 refacto folder counter + markRead when show details + small UI fix (#591)

mariusode committed:
fix: WB-3838, handle message content transformation with empty body (#592)

* fix: WB-3838, handle message content transformation with empty body

* fix: WB-3838, clean temp logging

* fix: WB-3838, fix after review

evanthonyly committed:
feat(conversation): #WB-3840 restore message (#593)

* Restore message action

* Refacto visibility icon condition

* Add test of useRestoreMessage

* Add toast message after restoring message

mariusode committed:
Feat wb 3837 original format existence (#595)

* feat: WB-3837, return data about original format message existence

* fix: WB-3837, missing privileges and fixing test

evanthonyly committed:
feat(conversation): #WB-3453 delete message (#599)

* Delete message action

* Add Test of delete message action

dboissin committed:
feat(conversation): WB-3498 remove sending block if quota is full

damienromito committed:
fix(conversation): get conversation/i18n translations file

damienromito committed:
feat(conversation): show "to" recipient list

damienromito committed:
feat(conversation): #WB-3881 display recipients list

juniorode committed:
fix(dev): #ENABLING-217, fetch modname from pre exported module version for js watch

evanthonyly committed:
feat(conversation): #WB-3826 display action list on message detail view

jcbe-ode committed:
chore: prepare develop-b2school version

Romu-C committed:
feat(auth): #WB-3888 international phone input auth module (#615)

mariusode committed:
Feat wb 3801 adapt legacy visibles (#613)

* chore: prepare develop-b2school version

* feat(communication): MOZO-206, new search visible API for messaging

* feat: MOZO-206 light version of search visibles

* feat: WB-3801, adapt legacy visible result to new api format

* feat: WB-3801, fix after merge

---------

Co-authored-by: Jean-Christophe <jean-christophe.benoit@opendigitaleducation.com>
Co-authored-by: Zakaria BANI <zakaria.bani@edifice.io>
Co-authored-by: Junior BERNARD <junior.bernard@opendigitaleducation.com>

Romu-C committed:
feat(conversation): #WB-3908 update message statut in list after opening details

Romu-C committed:
feat(conversation): #WB-3855 add unread action on message detail

mariusode committed:
feat: WB-3781, adding children data for year transition parent export (#612)

* chore: prepare develop-b2school version

* feat: WB-3781, adding children data for year transition parent export

---------

Co-authored-by: Jean-Christophe <jean-christophe.benoit@opendigitaleducation.com>

Romu-C committed:
feat(class-admin): #WB-3882 international number in user info page (#605)

evanthonyly committed:
Feat(conversation): WB-3457 use modal confirm on delete message (#616)

* Add ConfirmModal component

* Add useConfirmModalStore

* Display modal confirm before delete message on list

* Display confirm modal on delete list message

* Remove modal wrapper

* Change wording delete.definitely.confirm

* Use ConfirmModal from @edifice.io/react

* Add key of button action

* Move useConfirmModal on store/

Romu-C committed:
feat(class-admin): #WB-3732 add international mobile number in user relative creation

damienromito committed:
feat(conversation): #WB-3833 outbox details in message list (#614)

* chore: prepare develop-b2school version

* fix(conversation):  #WB-3833  add "at" before de recipient name

* fix(conversation):  #WB-3833  general fix nosubject text coloe

* fix(conversation):  #WB-3833  display recipient avatar in outbox

* fix(conversation):  #WB-3833  truncate recipient list only on message list

* fix(conversation):  #WB-3833  remove MessageFolderId type

* fix(conversation):  #WB-3833  setup code format

* fix(conversation):  #WB-3833  option to disable link on recipientList

* fix(conversation):  #WB-3833  remove cci recipients - thx Loic

* feat(conversation):  #WB-3833  add avatar group icon

* refactor(conversation):  #WB-3833  split logic from interface and apply best practices

* feat(conversation): in outbox list, display all recipients (to, cc, cci)

* fix(conversation): file naming

---------

Co-authored-by: Jean-Christophe <jean-christophe.benoit@opendigitaleducation.com>

evanthonyly committed:
Feat(conversation): WB-3878 restore action on detail view

damienromito committed:
fix(conversation):  #WB-3833   component declared but its value is never read

Romu-C committed:
fix(class-admin): #WB-3909 watch ngModel changes

Romu-C committed:
chore(conversation): filename convention + folder split for message and message list features (#618)

* fix useMessage when no messages liste

evanthonyly committed:
Move to trash on display message

damienromito committed:
feat(conversation): #WB-3881 show cci if current user is sender (#619)

jcbe-ode committed:
feat(common,directory): #WB-1404, framework extension for actualites app requirements (#622)

* feat(common): #WB-1404, add a hook for running post-SQL scripts tasks

* Add a method to get details on users' structures

Romu-C committed:
fix(conversation): #WB-3431 update unread count depending on messages selected status (#623)

* fix: #WB-3431 update unread count depending on messages selected status

* fine tuning read message

evanthonyly committed:
Display reply all only necessary in message detail view

damienromito committed:
feat(conversation): #WB-3834 draft list message preview (#620)

* feat(conversation): #WB-3834 draft - show recipient rather than sender in message list

* feat(conversation): #WB-3834 draft - show draft label before recipient list

* feat(conversation): #WB-3834 draft - hide "at" label before recipient list + improve accessibility

* feat(conversation): #WB-3834 draft - possible to have any recipients

* feat(conversation): #WB-3834 draft - add empty recipient icon

* fix(conversation): #WB-3834 draft - review

* fix(conversation): #WB-3834 draft - review - add system folder id

* fix(conversation): #WB-3834 draft - review

* fix(conversation): #WB-3834 draft -  add SystemFolder type for useSelectedFolder hook

jcbe-ode committed:
feat(conversation): #WB-3817,  endpoint to delete folders while trash… (#588)

* feat(conversation): #WB-3817,  endpoint to delete folders while trashing their messages

* feat : make frontend use the new endpoint

* fix: feedback from PR

* test: it tests

* fix: feedback from PR

Romu-C committed:
fix(conversation): #3431 draft message can not be unread or read

Romu-C committed:
test(conversation): #WB-3817 add missing mock API

Romu-C committed:
feat(conversation): #WB-3436 create new message component (#626)

* feat(conversation): #WB-3436 create new message component

* Add missing subject input + feedback correction

Romu-C committed:
feat(conversation): #WB-3437 save draft (#629)

damienromito committed:
feat(conversation):WB-3836 show origin folder icon (#627)

* feat(conversation): #WB-3836 display origin icon

* feat(conversation): #WB-3836 fix mock

* feat(conversation): #WB-3836 clean isInUserFolder condition

* feat(conversation): #WB-3836 display right origin folderId on user folder list

* feat(conversation): #WB-3836 optimise useMessageFolderId

damienromito committed:
feat(conversation):#WB-3426 inbox list message preview

commit b3135ca2ce3eda1a5ef75d51ca578229db32011a
Author: Damien Romito Castro <damien@romito.fr>
Date:   Fri Mar 14 15:35:33 2025 +0100

    fix(conversation):#WB-3426 small fixes

commit e9eaeba799f700d0576ae3d7e97f5f8302fe1dc1
Author: Damien Romito Castro <damien@romito.fr>
Date:   Fri Mar 14 11:19:44 2025 +0100

    fix(conversation):#WB-3426 add html semantic tag

commit ec3eef6e5f5b3ab4182d5ef6cef4757e3c5e4f09
Author: Damien Romito Castro <damien@romito.fr>
Date:   Thu Mar 13 18:11:53 2025 +0100

    fix(conversation):#WB-3426 fix sender link font in message details

commit 337b5f893a9fefab3cc46b54b76c88dfaef74bd6
Author: Damien Romito Castro <damien@romito.fr>
Date:   Thu Mar 13 17:53:33 2025 +0100

    fix(conversation):#WB-3426 fix cci to display with hook useMessageCciToDisplay

commit 152f8bf386d02ddf02b9b2df61ee9872a4744631
Author: Damien Romito Castro <damien@romito.fr>
Date:   Thu Mar 13 16:29:09 2025 +0100

    feat(conversation):#WB-3426 resize message date

commit 2ade00c55e35eb3ad9c0693fd64bf005fe9f3329
Author: Damien Romito Castro <damien@romito.fr>
Date:   Thu Mar 13 15:26:55 2025 +0100

    feat(conversation):#WB-3426 resize forwarded icon + clean accessibilty

damienromito committed:
feat(conversation):WB-3836 user folder id exception

commit 954bb947001e5541150c3ade756120fb97fedd7d
Author: Damien Romito Castro <damien@romito.fr>
Date:   Fri Mar 14 18:26:38 2025 +0100

    fix(conversation):#WB-3836 fix test wording

commit e9a2f6d44cf2b9a9b540da9964d5b8135ed13b0a
Author: Damien Romito Castro <damien@romito.fr>
Date:   Fri Mar 14 18:25:40 2025 +0100

    fix(conversation):#WB-3836 fix test wording

commit 5272ae16d5de0bb3e6ddd01ff51dede3f6708ff6
Author: Damien Romito Castro <damien@romito.fr>
Date:   Fri Mar 14 18:22:59 2025 +0100

    fix(conversation):#WB-3836 should display inbox icon when current user send the message to himself in user forlder

evanthonyly committed:
feat(conversation):WB-3452 move message to folder on list view (#633)

* Display move icon on message list header

* Add modal MoveMessageToFolderModal

* Display folders on modal

* Add hook useFolderUtils

* Display toast and redirect to folder after moving message

* Review

Romu-C committed:
fix(conversation): #WB-3431 update action accessibility (#631)

* add utils

* add test for display action in list

Romu-C committed:
fix(class-admin): #WB-3940 #WB-3941 saving mobil phone data

jcbe-ode committed:
feat(admin, directory, feeder): WB-3782, filtering parent or student manual groups with levels (#635)

* feat: WB-3782, filtering parent or student manual groups with levels

* feat: WB-3782, enriching endpoint fetching users' levels with inherit option

* Allow filtering BroadcastGroups' users having Student or Relative profile by levels

* fix: feeder with autolinkUsersFromLevels

* feedback from PR

---------

Co-authored-by: mariusestaque <marius.estaque@opendigitaleducation.com>
Co-authored-by: Damien Romito Castro <damien@romito.fr>

damienromito committed:
feat(conversation):#WB-3450 download attachement (#634)

* feat(conversation): #WB-3450 download attachement clean code

* feat(conversation): #WB-3450 download all attachements

* feat(conversation): #WB-3450 fix style + download urls

* fix(conversation):#WB-3450 fix review

* fix(conversation):#WB-3450 fix review - trads

damienromito committed:
feat(conversation):#WB-3450 download attachement (#634)

* feat(conversation): #WB-3450 download attachement clean code

* feat(conversation): #WB-3450 download all attachements

* feat(conversation): #WB-3450 fix style + download urls

* fix(conversation):#WB-3450 fix review

* fix(conversation):#WB-3450 fix review - trads

evanthonyly committed:
feat(conversation):#WB-3826 navigation bar and dropdown action list on message detail (#636)

* Add structure of MessageNavigation

* Distinction between top and bottom Action dropdown

* gap on structure message

evanthonyly committed:
feat(conversation):#WB-3461 remove from folder on view list (#637)

* Clean onSuccess of moveMessage

* Display remove from folder button on list

* Modal confirm remove from folder

* Use Promise allSettled onSuccess

damienromito committed:
fix(conversation):#WB-3450  attachement list missing key

damienromito committed:
feat(conversation):WB-3833 show cc in message preview (#638)

* fix(conversation):#WB-3833
css with gap in message

* fix(conversation):#WB-3833
use isInRecipient utils in  useMessageFolderId

* fix(conversation):#WB-3833
sort files and folders

* feat(conversation):#WB-3833
show cc and cci to message preview recipients

* fix(conversation):#WB-3833
add hooks folder

jcbe-ode committed:
fix(conversation): #WB-3455, re-route to inbox after deleting current folder

Romu-C committed:
feat(conversation): #WB-3444 auto save draft (#639)

* feat(conversation): #WB-3444 auto save draft

* add edit message page

* Fix navigation + body content + counter update

evanthonyly committed:
feat(conversation): #WB-3461 remove from folder on view detail (#641)

* Remove from folder on view detail

* Hide on dropdown bottom

Romu-C committed:
fix: #WB-3444 remove navigation + fix issue with editor

damienromito committed:
feat(conversation):WB-3835 trash list (#642)

* feat(conversation):#WB-3835
display message origin in trash

* fix(conversation):#WB-3835
rename  component MessagePreview as best practices

* fix(conversation):#WB-3835
rename  component RecipientList

* feat(conversation):#WB-3835
1d wording for recipient labels

* feat(conversation):#WB-3835
fix test

* feat(conversation):#WB-3835
fix test

evanthonyly committed:
feat(conversation): #WB-3452 move to folder on view detail

Romu-C committed:
chore: remove unused

Romu-C committed:
fix(conversation): #WB-3444 keep id after create

jcbe-ode committed:
feat(conversation): #WB-3925, add or remove attachments to draft messages (#644)

* feat(conversation): #WB-3925, add attachments to draft messages

* service testing

* wip

* wip

* todo: some renders fail

* tried fixing previous bug

* bug fixed

* Added mutating feedback to the user

* feat: Remove all attachments

* fix tests

* todo: attach files on a new draft do not render well

damienromito committed:
feat(conversation):WB-3957- empty trash (#647)

* feat(conversation):WB-3957- empty trash

* feat(conversation):WB-3957 clean reload List component with key

* refactor(conversation):WB-3957 create useToolbarActions hook

* refactor(conversation):WB-3957 rename MessageList.tsx

* refactor(conversation):WB-3957 create MessageItem component

* fix(conversation):WB-3957 manage List reloading for update global checkbox

* fix(conversation):WB-3957 fix List on frontend framwork to clean hack in MessageList

* refactor(conversation):WB-3957 add useToolbarVisibility hook

* refactor(conversation):WB-3957 reload list when searchparams change

* fix(conversation): #WB-3444 keep id after create

* feat(conversation): #WB-3925, add or remove attachments to draft messages (#644)

* feat(conversation): #WB-3925, add attachments to draft messages

* service testing

* wip

* wip

* todo: some renders fail

* tried fixing previous bug

* bug fixed

* Added mutating feedback to the user

* feat: Remove all attachments

* fix tests

* todo: attach files on a new draft do not render well

---------

Co-authored-by: RomuC <romuald.coulaud@gmail.com>
Co-authored-by: jcbe-ode <67910608+jcbe-ode@users.noreply.github.com>

damienromito committed:
fix(conversation):#WB-3833 fix abbr for cc or cci (#649)

jcbe-ode committed:
fix(conversation): #WB-3964, improved regex for the endpoint rendering the app's HTML (#651)

jcbe-ode committed:
feat(conversation, admin): #WB-3776, recall messages (#650)

* feat(conversation): #WB-3776, recall messages

* configuration service

* configuration hooks

* fix: select messages

* refactor: stay DRY

* removed unnecessary feature

* default values for parameters

* display recalled message

* parametrized max folder depth, limited to 5 anyway

* fix tests

* feedback from PR

* feedback from PR

jcbe-ode committed:
feat(conversation): #WB-3819, view messages in original format (#652)

damienromito committed:
fix(conversation):#WB-3447 add attachment to workspace #646 (#648)

* fix(conversation):#WB-3447 modal to add attachment to workspace

* fix(conversation):#WB-3447 use ff AddAttachmentToWorkspaceModal

* feat(conversation):#WB-3447 add FF AddAttachmentToWorkspaceModal translations

* feat(conversation):#WB-3447 add ff modal to add attachment to workspace

damienromito committed:
fix(conversation):#WB-3450 fix responsive css download attachement (#634)

jcbe-ode committed:
fix(conversation): #WB-3819, remove layout  of old-format messages

damienromito committed:
feat(conversation):#WB-3969 send attachement to user workspace (#653)

* feat(conversation):#WB-3969 move AddAttachementToWotkspaceModal from FF

* feat(conversation):#WB-3969 fix portal system for AddAttachementToWotkspaceModa

* feat(conversation):#WB-3969 download attachment blob

* feat(conversation):#WB-3969 upload attachement to workspace

damienromito committed:
feat(conversation):#WB-3969 send attachement to user workspace (#653)

* feat(conversation):#WB-3969 move AddAttachementToWotkspaceModal from FF

* feat(conversation):#WB-3969 fix portal system for AddAttachementToWotkspaceModa

* feat(conversation):#WB-3969 download attachment blob

* feat(conversation):#WB-3969 upload attachement to workspace

damienromito committed:
buid(conversation)/get-tokens-for-dev (#655)

* buid(conversation)/get-tokens-for-dev

* feat(conversation):#WB-3969 send attachement to user workspace (#653)

* feat(conversation):#WB-3969 move AddAttachementToWotkspaceModal from FF

* feat(conversation):#WB-3969 fix portal system for AddAttachementToWotkspaceModa

* feat(conversation):#WB-3969 download attachment blob

* feat(conversation):#WB-3969 upload attachement to workspace

jcbe-ode committed:
fix(conversation): #WB-3444, unwanted server requests issues

evanthonyly commented on conversation/frontend/src/services/queries/message.ts:
```diff
@@ -114,34 +115,36 @@ const useToggleUnread = (unread: boolean) => {
+            unread: !unreadFilter ? undefined : true,
+          }),
+          (data: InfiniteData<MessageMetadata>) => {
+            data.pages.forEach((page: any) => {
```
on peut typer page ?

evanthonyly commented on conversation/frontend/src/services/queries/folder.ts:
```diff
@@ -22,10 +22,15 @@ export const folderQueryOptions = {
-    return [...folderQueryOptions.base, folderId, 'messages', options] as const;
+    return [
+      ...folderQueryOptions.base,
+      folderId ?? 'undefined',
```
undefined en string ?

jcbe-ode replied:
Oui, c'est une query key. Ca permet de gérer `null` de la même manière que `undefined`.
Même si null ne devrait jamais se produire.

damienromito committed:
fix(conversation)/get-tokens-for-remove config.json

damienromito commented on conversation/frontend/src/routes/pages/MessageDisplay.tsx:
Quelle est la raison de changer `MessageDisplay` en `Message` ? Et dans ce cas pourquoi ne pas changer `FolderDisplay` en 'Folder`

jcbe-ode replied:
Car Message gère désormais à la fois "display" et "edit"
On pourrait aussi renommer FolderDisplay, en effet. Même s'il n'ya a pas de vue "edit" (c'est une modale)

damienromito commented on conversation/frontend/src/services/queries/folder.ts:
```diff
@@ -22,10 +22,15 @@ export const folderQueryOptions = {
 
   getMessagesQuerykey(
-    folderId: string,
+    folderId: string | null | undefined,
```
Du coup, ça m'étonne que `folderId` puisse être différent d'une string quand je regarde là où la fonction est appelé. Surtout maintenant qu'on utilise bien le hook `useSelectedFolder` ?

jcbe-ode replied:
Et pourtant, cela arrive !
Je ne sait pas pourquoi non plus, du coup la "programmation défensive" devrait nous aider.

damienromito replied:
je pense que c'est ça ` const { folderId } = useParams() as { folderId: string };`. quet tu as remplacé, 
Faut vraiment eviter de mettre `as type` ça surcharge. 
Or `const { folderId } = useParams()` => string|undefined

damienromito replied:
![Capture d’écran 2025-04-08 à 11 10 14](https://github.com/user-attachments/assets/b888d181-620d-49a0-88e5-6172a4cc2e2d)

damienromito replied:
je comprend pas non plus, il doit y avoir une surcharge de type quelque part

Romu-C committed:
feat(conversation): #WB-3636 add recipients to draft (#654)

* feat: #WB-3636 add recipients to draft

* Add default sharebookmark
Add empty search result message
Update with react query all calls for message edition

Romu-C committed:
feat(conversation): #WB-3978 broadcast group exception (#657)

Romu-C committed:
feat(conversation): #WB-3637 send message (#660)
